### DISCUSSION TITLE
Two beets plugins I wanted to share

### DISCUSSION THREAD

**Discussion by doronbehar** [beetbox/beets#4745] [Show and tell]
Links:

- [`mark-unmatched`](https://gitlab.com/doronbehar/beets-mark-unmatched)
- [`import_history`](https://github.com/doronbehar/beets-import_history)

### Motivation:

Sometimes when we import many directories in a bulk, we may encounter a directory that we can't / don't want to import because it's not listed on musicbrainz yet. The `mark-unmatched` plugin adds an option to the import menu - to add a `beets-unmatched` file to the directory, that can mark us to take care of this import later.

Here's how it looks:

![Screenshot from 2023-04-08 12-22-30](https://user-images.githubusercontent.com/10998835/230714141-0edd0067-ee8e-43a7-b72b-59faa81e63f5.png)

The 2nd plugin - `import_history`, adds a `source_path` to every file in the database, that tells you from where the file was imported. Again if you import files in a bulk, this can be useful, for example:

```
beet ls source_path:$PWD --format='$source_path' | awk -F/ '{print $((NF-1))}' | sort -u > imported.txt
```

I was thinking maybe it would be worth including these plugins in beets?

**Comment by sampsyo** [comment]
Sounds very cool! For starters, we can definitely link to the new plugins from our docs page that catalogs third-party packages.

As far as including them in beets: they seem simple and useful enough! Both seem like they could be candidates for inclusion. To make that work, they would basically just need to meet the usual standards of thorough documentation, etc.

**Reply by doronbehar** [reply]
> Sounds very cool! For starters, we can definitely link to the new plugins from our docs page that catalogs third-party packages.

Thanks! Do you think the WiKi could host such documentation? I think it fits this purpose more then the official documentation, since people like me could update it without waiting for official approval.

**Reply by sampsyo** [reply]
Good question—I don't have a strong objection to moving that stuff to the wiki, but one nice thing about keeping it in the "main" repo is that it keeps everything in one place. It's sorta nice to have stuff like that versioned. But I totally see your point about this list of external plugins wanting to be more of a free-for-all.

For the time being, any chance you could open a tiny PR and we'll go from there? Happy to keep discussing the possibility of moving the content on a parallel track.

**Comment by arsaboo** [comment]
Thanks for writing these plugins 👍

Please include them in the community plugin list so that these are not lost. Also, a little readme will be useful for folks looking at the repo.

**Comment by doronbehar** [comment]
The plugins are suggested here:

- https://github.com/beetbox/beets/pull/4748
- https://github.com/beetbox/beets/pull/4749
